data-structures.md

```diff
@@ -56,17 +81,25 @@ type Block struct {
 
 	// Height is the chain height of this block.
 	Height Uint64
-    
-    // StateRoot is a cid pointer to the state tree after application of the
+
+	// StateRoot is a cid pointer to the state tree after application of the
 	// transactions state transitions.
 	StateRoot Cid
 
-	// Messages is the set of messages included in this block
+	// ECDSAMessages is the set of messages signed with ECDSA included in this block
 	// TODO: should be a merkletree-ish thing
-	Messages []SignedMessage
+	ECDSAMessages []SignedMessage
+  
+  // BLSSignatureAggregate is the SignedMessage contianing the aggreate signature of 
+  // all messages signed with the BLS signature scheme.
+  BLSSignatureAggregate SignedMessage
+  
+  // BLSMessages is the set of messages with associated BLS signatures contained
+  // in BLSSignatureAggregate.
+  BLSMessages []Message
```

Comment:
In general I like this structure, but we are now unable to present an ordering of messages in the form of 

```
- bls message
- ecdsa message
- bls message
```
In regards to execution order. 

I would argue this is fine, and miners are simply asked to process these not together, but first one kind, then the other kind. But this is an explicit design decision that might have other implications.